services/apps/alcs/src/utils/search/notice-of-intent-search-filters.ts

```diff
@@ -0,0 +1,138 @@
+        },
+      )
+      .getMany();
+  },
+  addNameResults: (
+    searchDto: SearchRequestDto | InboxRequestDto,
+    noiSubmissionRepository: Repository<NoticeOfIntentSubmission>,
+  ) => {
+    const formattedSearchString =
+      formatStringToPostgresSearchStringArrayWithWildCard(searchDto.name!);
+    return noiSubmissionRepository
+      .createQueryBuilder('noiSub')
+      .select('noiSub.fileNumber')
+      .leftJoin(
+        NoticeOfIntentOwner,
+        'notice_of_intent_owner',
+        'notice_of_intent_owner.notice_of_intent_submission_uuid = noiSub.uuid',
+      )
+      .andWhere(
+        new Brackets((qb) =>
+          qb
+            .where(
+              "LOWER(notice_of_intent_owner.first_name || ' ' || notice_of_intent_owner.last_name) LIKE ANY (:names)",
+              {
+                names: formattedSearchString,
+              },
+            )
+            .orWhere(
+              'LOWER(notice_of_intent_owner.first_name) LIKE ANY (:names)',
+              {
+                names: formattedSearchString,
+              },
+            )
+            .orWhere(
+              'LOWER(notice_of_intent_owner.last_name) LIKE ANY (:names)',
+              {
+                names: formattedSearchString,
+              },
+            )
+            .orWhere(
+              'LOWER(notice_of_intent_owner.organization_name) LIKE ANY (:names)',
+              {
+                names: formattedSearchString,
+              },
+            ),
+        ),
+      )
+      .getMany();
+  },
+  addGovernmentResults: async (
+    searchDto: SearchRequestDto,
+    noiRepository: Repository<NoticeOfIntent>,
+    governmentRepository: Repository<LocalGovernment>,
+  ) => {
+    const government = await governmentRepository.findOneByOrFail({
+      name: searchDto.governmentName,
+    });
+
+    return noiRepository.find({
+      where: {
+        localGovernmentUuid: government.uuid,
+      },
+      select: {
+        fileNumber: true,
+      },
+    });
+  },
+  addParcelResults: (
+    searchDto: SearchRequestDto | InboxRequestDto,
+    noiSubmissionRepository: Repository<NoticeOfIntentSubmission>,
+  ) => {
+    const query = noiSubmissionRepository
+      .createQueryBuilder('noiSub')
+      .select('noiSub.fileNumber')
+      .leftJoin(
+        NoticeOfIntentParcel,
+        'parcel',
+        'parcel.notice_of_intent_submission_uuid = noiSub.uuid',
+      );
+
+    if (searchDto.pid) {
+      query.andWhere('parcel.pid = :pid', { pid: searchDto.pid });
+    }
+
+    if (searchDto.civicAddress) {
+      query.andWhere('LOWER(parcel.civic_address) like LOWER(:civic_address)', {
+        civic_address: `%${searchDto.civicAddress}%`.toLowerCase(),
+      });
+    }
+
+    return query.getMany();
+  },
+  addFileTypeResults: (
+    searchDto: InboxRequestDto,
+    noiRepository: Repository<NoticeOfIntent>,
+  ) => {
+    return noiRepository.find({
+      select: {
+        fileNumber: true,
+      },
+    });
+  },
```

Comment:
NOI also has decision components, should this be more like `APP_SEARCH_FILTERS.addFileTypeResults`?

Reply:
I don't know, this ticket was not about changing functionality only improving performance / maintainability. This filter works the same as before.